fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/properties/CardinalitiesProperty.java

```diff
@@ -368,6 +369,12 @@ public Cardinalities visitFullUnorderedScanExpression(@Nonnull final FullUnorder
         return Cardinalities.unknownCardinalities();
     }
 
+    @Nonnull
+    @Override
+    public Cardinalities visitGroupByExpression(@Nonnull final GroupByExpression element) {
+        return Cardinalities.unknownCardinalities();
```

Comment:
You should at least do the case where you have no grouping columns or a constant grouping column and set:
- `min` and `max` to `1` if you don't have grouping columns (greeting from the count bug)
- `min` to `0` and `max` to `1` if you have a constant grouping expression

Reply:
Great observation, thank you!

Reply:
Regarding the second case (constant grouping expression). I will take care of it once I am able to bring https://github.com/FoundationDB/fdb-record-layer/pull/1819 in since it provides an easy way of checking whether a `Value` can be evaluated at compile time (I am assuming compile-time === constant although this is generally not true, e.g. `rand()`, but I think that's a separate work stream). Do you agree?

Reply:
👍